### PR TITLE
Template and in-memory sample for Freyja apps

### PR DESCRIPTION
**Pull request by wilyle**
Adds a template and in-memory sample for Freyja apps. Also includes some initial docs and workflows for the repo

### CODE REVIEW

File: freyja_apps/in_memory/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# In-Memory Freyja Example Application
+
+This Freyja Example Application utilizes the in-memory mocks provided by Freyja to demonstrate a minimal example. When you start the application there will initially be no data flowing, but over time signals will get added to the mock digital twin and mapping client. Data is "emitted" by printing to stdout.
+
+## Build and Run
+
+To build and run the application, run the following from the repo root:
+
+```shell
+cargo build -p freyja-in-memory-app
```
Comment (ashbeitz) [inline freyja_apps/in_memory/README.md:10]: You can condense this down to just the run command.

File: freyja_apps/in_memory/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+cargo run -p freyja-in-memory-app
+```
+
+Note that running `cargo build` without specifying the `-p` argument will build every project in the workspace! When working with this repository it's recommended to use the `-p` argument with Cargo commands.
```
Comment (ashbeitz) [inline freyja_apps/in_memory/README.md:14]: replace the "build" with "run"

File: freyja_apps/template/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Freyja App Template
+
+This is a template for creating a Freyja application. For more information about writing Freyja applications, see the [Freyja documentation on the subject](https://github.com/eclipse-ibeji/freyja/blob/main/docs/custom-adapters.md).
+
+## Instructions
+
+To edit this template and create a Freyja application, follow these steps:
+
+1. Choose the adapters to use. Freyja requires users to select a Cloud Adapter, a Digital Twin Adapter, and a Mapping Client. Some potential choices are:
+    1. Freyja mocks. These are enumerated [here](https://github.com/eclipse-ibeji/freyja/blob/main/docs/quickstart.md#appendix-a) and are generally only suitable for tests or demos.
```
Comment (ashbeitz) [inline freyja_apps/template/README.md:10]: Please place periods at the end of each of the numbered lines.

File: freyja_apps/template/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+1. Edit `Cargo.template.toml`:
+    1. Change the `rev` property of the `freyja` dependency to match the commit hash of the Freyja version you want to use
+    1. Add dependencies for the package(s) you need for your adapters. You can implement adapters in the same crate and import their dependencies here, or you can implement them in separate crates and import those.
+    1. Update the package name and version as desired
+    1. Rename the file to `Cargo.toml` so that Cargo will be able to find it
+1. Edit `src/main.rs`:
+    1. Import your adapters with `use` statements
+    1. Change the type names in the `freyja_main!` macro invocation to your imported types. Order matters here, so make sure you list your adapters in the same order as they are presented in the template.
+
+To build and run your application, you can use Cargo commands such as `cargo build` and `cargo run`.
```
Comment (ashbeitz) [inline freyja_apps/template/README.md:22]: No need for cargo build.

**Review by ashbeitz — COMMENTED** [review]
Please take a look at my comments.

**Review by ashbeitz — APPROVED** [review]
Approved

**Review by jorchiu — APPROVED** [review]

File: freyja_apps/template/Cargo.template.toml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: MIT
+
+[package]
+name = "freyja-template-app"
+version = "0.1.0"
+edition = "2021"
+license = "MIT"
+
+[dependencies]
+# These two dependencies are required for anyone implementing a Freyja application
```
Comment (ladatz) [inline freyja_apps/template/Cargo.template.toml:12]: Similar to my other comment: consider adding a comment about how to choose the commit hash here

Comment (ladatz) [reply freyja_apps/template/Cargo.template.toml:12]: Ah, now I see it's in the documentation. Up to you if you want to add it here too

**Review by ladatz — APPROVED** [review]
Approving with just one suggestion

File: freyja_apps/template/Cargo.template.toml (hunk 5 above)
Comment (wilyle) [reply freyja_apps/template/Cargo.template.toml:12]: I'll leave it to the docs

**Pull request merged**